tests/nested/manual/core20-early-config/task.yaml

```diff
@@ -0,0 +1,69 @@
+summary: Test that config defaults are applied early.
+
+systems: [ubuntu-20.04-64]
+
+prepare: |
+    ENABLE_TPM="true"
+    ENABLE_SECURE_BOOT="true"
+    BUILD_SNAPD_FROM_CURRENT="true"
+    
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    cleanup_nested_env
```

Comment:
it would be nice if we had a different suite for these kinds of tests that really shouldn't have the nested VM prepared ahead of time before the test runs and instead just essentially allocate the system and let the test itself preparation manually build the image it needs to build and then boot it.

Maybe we rename the current `nested/*` suites to `nested-auto/*` as those are where the nested VM is automatically created, and we introduce a new suite just called nested (or maybe nested-manual, but here manual would have a double meaning and we could have tests named like nested-manual/manual/core20-early-config, which is confusing) where the nested VM is not automatically booted and instead is left to the test to be created/booted.

Reply:
It generally works this way already (but maybe I misunderstand your suggestion): normal _nested_ tests have a separate suite that manages the lifetime of VMs; and we have another suite for _nested/manual_ tests where VMs need to be managed from the test (I agree _manual_ is not the best name and slightly confusing, perhaps we should rename it). The management is done by helper functions such as create_nested_core_vm or start_nested_core_vm.

Now, if you're suggesting that nested/manual suite tests could have 'prepare' step left for the test but otherwise boot the VM automatically, then the problem is we need full control over starting the VMs in some tests (namely: preseeding tests where execute: part starts with snap-preseed, after which comes start_nested_core_vm). Therefore I think small helpers are most flexible, even if they lead to some duplication.

Reply:
ah you know what I had completely misread what tests/nested/manual was and how it was setup in the spread.yaml, I had thought I read it correctly in the spread.yaml, but I had actually just been reading the tests/nested/core test, my apologies. 

Since that this is the case, what is the purpose of running this in the prepare section? Would anything actually have created the work dir before the prepare section runs for this test?

Reply:
Fair point, I thoughtlessly copied it from another test, this shouldn't be needed as you noted (unless we encounter leftovers from previous test, which should be addressed elsewhere). I can see how this was confusing now. Removed. Thanks!